src/Turbo/Doctrine/BroadcastListener.php

```diff
@@ -0,0 +1,117 @@
+ *
+ * @see https://github.com/api-platform/core/blob/master/src/Bridge/Doctrine/EventListener/PublishMercureUpdatesListener.php Adapted from API Platform.
+ *
+ * @todo backport MongoDB support
+ *
+ * @experimental
+ */
+final class BroadcastListener implements ResetInterface
+{
+    private $broadcaster;
+
+    /**
+     * @var \SplObjectStorage<object, object>
+     */
+    private \SplObjectStorage $createdEntities;
+    /**
+     * @var \SplObjectStorage<object, object>
+     */
+    private \SplObjectStorage $updatedEntities;
+    /**
+     * @var \SplObjectStorage<object, object>
+     */
+    private \SplObjectStorage $removedEntities;
+
+    public function __construct(BroadcasterInterface $broadcaster)
+    {
+        if (80000 > \PHP_VERSION_ID) {
+            throw new \LogicException('The broadcast feature requires PHP 8.0 or greater, you must either upgrade to PHP 8 or disable it.');
+        }
+
+        $this->reset();
+
+        $this->broadcaster = $broadcaster;
+    }
+
+    /**
+     * Collects created, updated and removed entities.
+     */
+    public function onFlush(EventArgs $eventArgs): void
+    {
+        if (!$eventArgs instanceof OnFlushEventArgs) {
+            return;
+        }
+
+        $uow = $eventArgs->getEntityManager()->getUnitOfWork();
+        foreach ($uow->getScheduledEntityInsertions() as $entity) {
+            $this->storeEntitiesToPublish($entity, 'createdEntities');
+        }
+
+        foreach ($uow->getScheduledEntityUpdates() as $entity) {
+            $this->storeEntitiesToPublish($entity, 'updatedEntities');
+        }
+
+        foreach ($uow->getScheduledEntityDeletions() as $entity) {
+            $this->storeEntitiesToPublish($entity, 'removedEntities');
+        }
+    }
+
+    /**
+     * Publishes updates for changes collected on flush, and resets the store.
+     */
+    public function postFlush(): void
+    {
+        try {
+            foreach ($this->createdEntities as $entity) {
+                $this->broadcaster->broadcast($entity, Broadcast::ACTION_CREATE);
+            }
+
+            foreach ($this->updatedEntities as $entity) {
+                $this->broadcaster->broadcast($entity, Broadcast::ACTION_UPDATE);
+            }
+
+            foreach ($this->removedEntities as $entity) {
+                $this->broadcaster->broadcast($entity, Broadcast::ACTION_REMOVE);
+            }
+        } finally {
+            $this->reset();
+        }
+    }
+
+    public function reset(): void
+    {
+        $this->createdEntities = new \SplObjectStorage();
+        $this->updatedEntities = new \SplObjectStorage();
+        $this->removedEntities = new \SplObjectStorage();
+    }
+
+    private function storeEntitiesToPublish(object $entity, string $property): void
+    {
+        if ((new \ReflectionClass($entity))->getAttributes(Broadcast::class)) {
+            $this->{$property}->attach('removedEntities' === $property ? clone $entity : $entity);
```

Comment:
Clone can be pretty expensive for an entity and cause weird behavior. I see in `book.stream.html.twig` this is used to keep the entity id since its necessary to build the correct `target`. I wish there was a better solution to this.

Reply:
As other parts of the bundle require to have a `$id` property or a `getId()` method, maybe should we use this convention here too and just store the id?

Reply:
I just looked at this, unfortunately this will hurt the user experience. The current implementation allows accessing to the whole removed entity in the template. Storing only the ID will prevent to do this. I suggest keeping this as-is for now, until we find a better solution.

Reply:
Fixed with the new `IdAccessor`.